framework/src/bundle/BundleContext.cpp

```diff
@@ -94,6 +102,10 @@ Any BundleContext::GetProperty(const std::string& key) const
 
 AnyMap BundleContext::GetProperties() const
 {
+  if (!d) {
+    throw std::runtime_error("The bundle context is no longer valid");
```

Comment:
The doxygen comment in `BundleContext.h` needs to be updated to indicate this now throws.